java/fury-core/src/main/java/org/apache/fury/codegen/CodegenContext.java

```diff
@@ -285,7 +288,26 @@ public String type(Class<?> clz) {
     String type = ReflectionUtils.getCanonicalName(clz);
     if (type.startsWith("java.lang")) {
       if (!type.substring("java.lang.".length()).contains(".")) {
-        return clz.getSimpleName();
+        String simpleName = clz.getSimpleName();
+        boolean hasPackage = StringUtils.isNotBlank(pkg);
+        Map<String, Boolean> packageMap =
+            nameConflicts.computeIfAbsent(hasPackage ? pkg : "", p -> new ConcurrentHashMap<>());
+        Boolean conflictRes =
+            packageMap.computeIfAbsent(
+                simpleName,
+                sn -> {
+                  try {
+                    ClassLoader beanClassClassLoader =
+                        clz.getClassLoader() == null
+                            ? Thread.currentThread().getContextClassLoader()
+                            : clz.getClassLoader();
+                    beanClassClassLoader.loadClass(hasPackage ? pkg + "." + sn : sn);
+                    return Boolean.TRUE;
+                  } catch (ClassNotFoundException e) {
+                    return Boolean.FALSE;
+                  }
+                });
+        return conflictRes.booleanValue() ? clz.getName() : simpleName;
```

Comment:
```suggestion
        return conflictRes ? clz.getName() : simpleName;
```

Unbox is unnecessary